### ISSUE TITLE
Automatically generate link_name attributes in boringssl.rs

### ISSUE THREAD

**Issue by joshlf** [google/mundane#7]
Currently, our `bindgen.sh` script does not automatically generate the `link_name` attributes that we need in order to perform symbol renaming as described in `boringssl/README.md`. Instead, we do this manually, which is annoying.

We should do this automatically. One good option would be to implement the `bindgen` feature described in this issue: https://github.com/rust-lang-nursery/rust-bindgen/issues/1375

**Comment by inejge** [comment]
Can a workaround be a postprocessing script called from `bindgen.sh`? I tested the following on a copy of `boringssl.rs` with `#[link name...]` lines removed and got the original file back. I can open a PR if the script is good enough.

```diff
diff --git a/boringssl/bindgen.sh b/boringssl/bindgen.sh
index 564423c..e52c289 100755
--- a/boringssl/bindgen.sh
+++ b/boringssl/bindgen.sh
@@ -143,3 +143,16 @@ cat boringssl.rs >> "$TMP"
 
 mv "$TMP" boringssl.rs
 rustfmt boringssl.rs
+
+awk -v "vers=${MAJOR}_${MINOR}_${PATCH}_" '
+/extern "C" {/ {
+    print
+    getline
+    if ($0 ~ "pub fn") {
+        fn = $3
+        sub("[(].*", "", fn)
+        print "    #[link_name = \"__RUST_MUNDANE_" vers fn "\"]"
+    }
+}
+{ print }' boringssl.rs >> "$TMP"
+mv "$TMP" boringssl.rs
```

**Comment by joshlf** [comment]
Yeah, that looks awesome! Check out `CONTRIBUTING.md` to see how to submit it. Would it be possible to also count the number of `link_name` attributes emitted by the script? We could make sure it matches the number of functions we expect in the script as a sanity check. I'm always a little wary of using simple syntax analysis on code because it can be so error-prone.

**Comment by joshlf** [comment]
Btw if you want to work on this, it'll probably be easier for you if you work on top of [this PR](https://fuchsia-review.googlesource.com/c/mundane/+/227173) (which is in turn on top of [this PR](https://fuchsia-review.googlesource.com/c/mundane/+/227172)).

**Comment by inejge** [comment]
[Uploaded](https://fuchsia-review.googlesource.com/c/mundane/+/227076), let's hope I haven't messed up anything in the procedure.

**Comment by joshlf** [comment]
Looks like you got all of the Gerrit magic right; well done! Faster than it took me to figure out for sure.

**Comment by inejge** [comment]
(Signing off for now; I saw the comments & will reply later)
